code/modules/clothing/head/misc_special.dm

```diff
@@ -152,7 +152,28 @@
 	armor = list(MELEE = 0, BULLET = 0, LASER = 0,ENERGY = 0, BOMB = 0, BIO = 0, RAD = 0, FIRE = 0, ACID = 0)
 	light_range = 2 //luminosity when on
 	flags_cover = HEADCOVERSEYES
+	light_color = "#fff2bf"
+	worn_y_offset = 1
 
+/obj/item/clothing/head/hardhat/pumpkinhead/update_overlays()
+	. = ..()
+	cut_overlays()
+	if(light_on)
+		. += emissive_appearance(icon, "carved_pumpkin-emissive", alpha = src.alpha)
+
+/obj/item/clothing/head/hardhat/pumpkinhead/worn_overlays(isinhands)
+	. = ..()
+	var/static/on_overlay
+	if(!isinhands)
+		. += emissive_appearance(worn_icon, "carved_pumpkin-emissive", alpha = src.alpha)
```

Comment:
```suggestion
		. += emissive_appearance(worn_icon, "carved_pumpkin-emissive", alpha = src.alpha, appearance_flags = KEEP_APART)
```
Because BYOND is maximum jank this is required for emissive overlays on worn overlays. Essentially, because humans have `KEEP_TOGETHER` it tries to squish all of the overlays together and this messes with the layering. `KEEP APART` lets overlays exempt from this this and overlays with different planes are treated as having `KEEP_APART`... if they are directly applied to the mob. Since this is an overlay on an overlay BYOND doesn't notice the separate plane and we need to explicitly declare this or it will smoosh them together anyways.